client/packages/system/src/Encounter/ListView/columns.ts

```diff
@@ -9,13 +9,20 @@ import { useFormatDateTime } from '@common/intl';
 import {
   ProgramEventFragment,
   EncounterRowFragment,
+  useDocumentRegistry,
 } from '@openmsupply-client/programs';
 
 const encounterEventCellValue = (events: ProgramEventFragment[]) => {
   // just take the name of the first event
   return events[0]?.data ?? '';
 };
 
+const getProgramName = (program: string) => {
+  const { data } = useDocumentRegistry.get.documentRegistryByType(program);
+
+  return data?.[0]?.name;
+};
+
```

Comment:
This probably works and is legal but I find it hard reason about it... What I mean is that the hook is hidden in this function but we have to make sure to always call all hooks to not confuse React. Possible problem is that you may return early from `useEncounterListColumns` before calling `getProgramName`. That this is not allowed is not obvious, e.g. for a different dev...

I would just inline this method, i.e. call `useDocumentRegistry.get.documentRegistryByType` straight after `useFormatDateTime`....

Reply:
[90e5fa1](https://github.com/openmsupply/open-msupply/pull/1417/commits/90e5fa118615508d4be9b843da99d509e6e69816)